internal/ledgerutil/compare.go

```diff
@@ -23,9 +23,15 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	// AllDivsByKey - Filename for the json output that contains all divergences ordered by key
+	AllDivsByKey = "all_divergences_by_key.json"
+)
+
 // Compare - Compares two ledger snapshots and outputs the differences in snapshot records
-// This function will overwrite the file at outputPath if it already exists
-func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count int, err error) {
+// This function will throw an error if the output directory already exist in the outputDirLoc
+// Function will return count of -1 if the public state hashes are the same
```

Comment:
If there are 0 differences, wouldn't the logical count to return be 0? Is there a reason to use the magic value of -1 instead of the logical value of 0?
You were probably trying to differentiate between the error case which also returns 0. But in the error case you can check for existence of error returned rather than rely on the count.

Reply:
I thought we decided we weren't going to throw an error if the public state hashes were the same. If the user is simply checking to see if two snapshots are the same, and they end up being the same, this technically shouldn't be an error right?

Reply:
Right, no error, I'm simply suggesting if they are the same and no errors, the logical way to reflect that would be to return count of 0 rather than -1.

Reply:
Ah, I see what you were thinking now... you want to distinguish between no comparison done and (the unexpected) comparison done and 0 differences found... in that case -1 response is ok.